public/app/core/table_model.ts

```diff
@@ -40,8 +41,19 @@ export default class TableModel {
     this.rows.sort((a, b) => {
       a = a[options.col];
       b = b[options.col];
+
+      if (kbn.isNumeric(a)) {
```

Comment:
I think it would be better to use the column type to decide how to sort values.
I'm not sure how this would work when strings and numeric values are mixed.

Reply:
@bergquist thanks for the feedback.

1) I think it would be better to use the column type to decide how to sort values.
> Are you referring to the property `columns` of `TableModel`? 

It is defined as optional so it may not be reliable. Looking at the datasource for `influxdb`, the code only ever populates the `time` type. 

2) I'm not sure how this would work when strings and numeric values are mixed.
> This one is interesting.

I had a look at the test suite and there isn't one for it. So I'm not sure how it should react at the moment. 

I'm thinking maybe the numeric conversion should only occur when both data type matches? I'll have a look again tonight when I get time. 

Alternatively, we can possibly re-work on the sort algorithm a bit. Instead of conquering everything in one line, maybe it should loop through the array to pluck out all NULLs, undefined, string and number into individually then for each array, sort them seperately and merge them back into a long list.

The one liner sorting algorithm looks extremely scary at the moment, and it is difficult for the next dev to debug.

Reply:
@bergquist Hey Carl, I have reworked on the sort functionality as described above.

This means that when a column has a mixture of string and number value. Numbers / stringified number will always come before string value. 

This is an improvement from previous implementation where there is no guarantee ordering when sorting column with string and number.

I have also added tests to show case this behaviour. Please have a look.